### DISCUSSION TITLE
Consulta recebimento de QRCode estático

### DISCUSSION THREAD

**Discussion by Eliandromj** [bacen/pix-api#465] [Q&A]
Olá, Estou fazendo a integração de meu ERP com API PIX do PagSeguro.

Estou gerando o QRCode de forma estática, validei, efetuei o pagamento tudo certo sem problemas a princípio.  (Tudo isso em ambiente de produção)

Ocorre que não estou conseguindo consultar o recebimento destes pagamentos pelas APIs "Consultar PIX" e "Consultar PIX recebidos".
Entendo que por ser estático não há registro do mesmo feito anteriormente, porém no manual de iniciação do pix (II_ManualdePadroesparaIniciacaodoPix-versao2-6-0.pdf -  Pág 7 - Item 1.5.5)  contém o seguinte texto:

_"Apesar de a criação do QR Code estático não fazer parte das funcionalidades da API Pix, os Pix
recebidos que tenham sido iniciados via QR Codes estáticos que estejam associados a um txid podem
ser consultados via API Pix. Maiores detalhes sobre essa funcionalidade podem ser encontrados no
repositório oficial da API Pix no perfil do BCB no Github: https://github.com/bacen/pix-api."_

Logo, ao meu ver eu deveria conseguir consultar estes recebimentos feitos através de QRCodes estáticos.
Porém como é a primeira vez que estou fazendo isso, não sei muito ao certo.

Gostaria de uma ajuda se alguém trabalhar desta forma (QRCode estático) e onde eu poderia estar errando.

Grato.

**Comment by rubenskuhl** [comment]
PagSeguro ou BoaCompra ? Ambos não tem bom histórico de seguir a API padrão...
De qualquer forma, na geração do estático você colocou um txid ?

**Reply by Eliandromj** [reply]
PagSeguro. 
Sim coloquei o txID, inclusive quando efetuei o pagamento ficou certo no recibo.
Os endpoitns que utilizei para tentar consultar os recebimento foram:

https://secure.api.pagseguro.com/instant-payments/pix/{e2eid} (Onde este e2eid peguei quando fiz o pagamento já que o QRCode é estático)

Response: 
{
    "status": 404,
    "detail": "Pix não encontrado para o e2eid informado.",
    "type": "https://pix.bcb.gov.br/api/v2/error/PixNaoEncontrado",
    "title": "Pix não encontrado",
    "error_messages": [
        {
            "code": "nao_encontrado",
            "description": "Pix não encontrado para o e2eid informado."
        }
    ]
}

https://secure.api.pagseguro.com/instant-payments/pix?fim=2021-11-04T15:25:59.411Z&inicio=2021-11-03T08:25:59.411Z (

Response:

{
    "parametros": {
        "inicio": "2021-11-03T08:25:59Z",
        "fim": "2021-11-04T15:25:59Z",
        "paginacao": {
            "paginaAtual": 0,
            "itensPorPagina": 100,
            "quantidadeDePaginas": 0,
            "quantidadeTotalDeItens": 0
        }
    },
    "pix": []
}
